WcaOnRails/app/models/light_result.rb

```diff
@@ -32,34 +33,26 @@ def initialize(r)
     @best = r["best"]
     @average = r["average"]
     @personName = r["personName"]
-    @eventId = r["eventId"]
-    @formatId = r["formatId"]
-    @roundId = r["roundId"]
     @pos = r["pos"]
     @personId = r["personId"]
     @regionalSingleRecord = r["regionalSingleRecord"]
     @regionalAverageRecord = r["regionalAverageRecord"]
-    @countryId = r["countryId"]
+    @country = Country.find(r["countryId"])
+    @format = Format.find(r["formatId"])
+    @round = Round.find(r["roundId"])
+    @event = Event.find(r["eventId"])
```

Comment:
Lazy as in "queries the database if the cache is empty"? Then no, that's going over the network + mutating global state at least some of the times.